pycbc/workflow/configuration.py

```diff
@@ -29,65 +29,183 @@
 import itertools
 import pycbc_glue.pipeline
 
+from cookielib import (_warn_unhandled_exception, LoadError, Cookie)
+from bs4 import BeautifulSoup
+
+def _really_load(self, f, filename, ignore_discard, ignore_expires):
+    """
+    This function is required to monkey patch MozillaCookieJar's _really_load
+    function which does not understand the curl format cookie file created
+    by ecp-cookie-init. It patches the code so that #HttpOnly_ get loaded.
+
+    https://bugs.python.org/issue2190
+    https://bugs.python.org/file37625/httponly.patch
+    """
+    now = time.time()
+
+    magic = f.readline()
+    if not re.search(self.magic_re, magic):
+        f.close()
+        raise LoadError(
+            "%r does not look like a Netscape format cookies file" %
+            filename)
+
+    try:
+        while 1:
+            line = f.readline()
+            if line == "": break
+
+            # last field may be absent, so keep any trailing tab
+            if line.endswith("\n"): line = line[:-1]
+
+            sline = line.strip()
+            # support HttpOnly cookies (as stored by curl or old Firefox).
+            if sline.startswith("#HttpOnly_"):
+                line = sline[10:]
+            # skip comments and blank lines XXX what is $ for?
+            elif (sline.startswith(("#", "$")) or sline == ""):
+                continue
+
+            domain, domain_specified, path, secure, expires, name, value = \
+                    line.split("\t")
+            secure = (secure == "TRUE")
+            domain_specified = (domain_specified == "TRUE")
+            if name == "":
+                # cookies.txt regards 'Set-Cookie: foo' as a cookie
+                # with no name, whereas cookielib regards it as a
+                # cookie with no value.
+                name = value
+                value = None
+
+            initial_dot = domain.startswith(".")
+            assert domain_specified == initial_dot
+
+            discard = False
+            if expires == "":
+                expires = None
+                discard = True
+
+            # assume path_specified is false
+            c = Cookie(0, name, value,
+                       None, False,
+                       domain, domain_specified, initial_dot,
+                       path, False,
+                       secure,
+                       expires,
+                       discard,
+                       None,
+                       None,
+                       {})
+            if not ignore_discard and c.discard:
+                continue
+            if not ignore_expires and c.is_expired(now):
+                continue
+            self.set_cookie(c)
+
+    except IOError:
+        raise
+    except Exception:
+        _warn_unhandled_exception()
+        raise LoadError("invalid Netscape format cookies file %r: %r" %
+                        (filename, line))
+
+# Now monkey patch the code
+cookielib.MozillaCookieJar._really_load = _really_load
```

Comment:
This raises a landscape warning https://landscape.io/github/duncan-brown/pycbc/145/modules/pycbc/workflow/configuration.py#L124. As this seems needed, please tell landscape to explicitly ignore this:
https://docs.landscape.io/suppressing.html
using the "suppress a single error" instructions.